<launch>
	<arg name="pid_rate" default="100" />
	<arg name="kp" default="0" />
	<arg name="ki" default="0" />
	<arg name="kd" default="0" />
	<arg name="upper_limit" default="2000" />
	<arg name="lower_limit" default="-2000" />
	<arg name="anti_windup" default="0.1" />
	<arg name="topic_pid_in" default="auvsi/pid/in" />
	<arg name="topic_pid_out" default="auvsi/pid/out" />
	<!-- topic for kp,kd, ki /auvsi/pid/const -->
	
	<node name="simple_red_controller" pkg="roboboat" output="screen" type="simple_red_controller" />
	<node name="avoid_controller" pkg="roboboat" output="screen" type="avoid_controller" />
	
	<node name="control" pkg="pid" output="screen" type="controller_adaptive" args="$(arg kp) $(arg ki) $(arg kd) $(arg pid_rate) -tfp $(arg topic_pid_in) -tfc $(arg topic_pid_out) -ul $(arg upper_limit) -ll $(arg lower_limit) -aw $(arg anti_windup)" />
	
	<node name="image_front" pkg="roboboat" output="screen" type="image_front" />
	<node name="red_hough" pkg="roboboat" output="screen" type="red_hough" />
	<node name="green_hough" pkg="roboboat" output="screen" type="green_hough" />
	
	<node name="motor_controller" pkg="roboboat" output="screen" type="motor_controller" />
	<node name="node_controller" pkg="roboboat" output="screen" type="node_controller" />
	<node name="remote_monitor" pkg="roboboat" output="screen" type="remote_monitor" />
	<node name="debug_makara" pkg="roboboat" output="screen" type="debug_makara" />
</launch>
